src/PowerShellEditorServices/Workspace/ScriptFile.cs

```diff
@@ -311,12 +311,34 @@ public void ValidatePosition(BufferPosition bufferPosition)
         /// <param name="column">The 1-based column to be validated.</param>
         public void ValidatePosition(int line, int column)
         {
-            int maxLine = this.FileLines.Count;
+            ValidatePosition(line, column, isInsertion: false);
+        }
+
+        /// <summary>
+        /// Throws ArgumentOutOfRangeException if the given position is outside
+        /// of the file's buffer extents.
```

Comment:
Perhaps we should add some explanation here to why column-max-index+1 and line-max-index+1 are allowed.